src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/GitFileManager.cs

```diff
@@ -806,5 +885,96 @@ void BuildDependencies(XmlNodeList dependencies)
 
             return dependencyDetails.Where(d => !d.Pinned);
         }
+
+        private HashSet<string> FlattenLocations(Dictionary<string, HashSet<string>> assetLocationMap, Func<string, bool> filter = null)
+        {
+            HashSet<string> managedFeeds = new HashSet<string>();
+
+            foreach (HashSet<string> locations in assetLocationMap.Values)
+            {
+                IEnumerable<string> filteredLocations = filter != null ?
+                    locations.Where(filter) :
+                    locations;
+
+                managedFeeds.UnionWith(filteredLocations);
+            }
+            return managedFeeds;
+        }
+
+        private List<(string key, string feed)> GetPackageSources(XmlDocument nugetConfig, Func<string, bool> filter = null)
+        {
+            var sources = new List<(string key, string feed)>();
+            XmlNodeList nodes = nugetConfig.SelectNodes("//configuration/packageSources/add");
+            foreach (XmlNode node in nodes)
+            {
+                if (node.NodeType != XmlNodeType.Comment && node.NodeType != XmlNodeType.Whitespace)
```

Comment:
Nit: isn't there an specific NodeType that you want to handle here?

Reply:
Yeah, I can probably just check if it's an element.